test/abc/sound/SequencePlayerTest.java

```diff
@@ -1,38 +1,125 @@
 package abc.sound;
 
-import static org.antlr.v4.runtime.CharStreams.fromFileName;
+import javax.sound.midi.InvalidMidiDataException;
+import javax.sound.midi.MidiUnavailableException;
 
-import java.io.IOException;
-
-import org.antlr.v4.runtime.CharStream;
-import org.antlr.v4.runtime.CommonTokenStream;
-import org.antlr.v4.runtime.tree.ParseTree;
-import org.antlr.v4.runtime.tree.ParseTreeWalker;
 import org.junit.Test;
 
-import abc.parser.AbcBaseListener;
-import abc.parser.AbcLexer;
-import abc.parser.AbcParser;
-
 public class SequencePlayerTest {
 
 	// TODO: warmup #2
 
 	@Test
-	public void test() {
-		String fileName = "sample_abc/piece1.abc";
+	public void piece1() {
+		try {
+			SequencePlayer player = new SequencePlayer(140, 12);
```

Comment:
Why did you select 12 for `ticksPerBeat` parameter?

Reply:
Yup. It's to speed up the song. If I have the ticksPerBeat parameter as 4, then it's very slow making both the pieces to collide and play together because of the threads.

Reply:
If you use the dynamic sleep the threads don't run into each other like that.